submissions/ses3332002/movie-seat-booking-js/script.js

```diff
@@ -0,0 +1,163 @@
+        break;
+      case 2:
+        rowString += `<input type="checkbox" id="seat${rowIndex + 1}_${i + 1}" disabled checked /><label for="seat${
+          rowIndex + 1
+        }_${i + 1}" class="seat"></label>`;
+        break;
+      case 3:
+        rowString += `<input type="checkbox" id="seat${rowIndex + 1}_${i + 1}" data-type="vip"><label for="seat${
+          rowIndex + 1
+        }_${i + 1}" class="seat" data-price="${vipSeatPrice}" /></label>`;
+        break;
+      case 4:
+        rowString += `<input type="checkbox" id="seat${rowIndex + 1}_${
+          i + 1
+        }" disabled checked data-type="vip"/><label for="seat${rowIndex + 1}_${i + 1}" class="seat"/></label>`;
+    }
+  });
+  return rowString;
+}
+
+function initAuditorium() {
+  let renderString = '';
+  seats.forEach((item, i) => {
+    renderString += `<div class="auditorium__row">${prepareRow(item, i)}</div>`;
+  });
+  renderString += '<input type="submit" class="seats_submit" value="Заказать">';
+  seatsForm.innerHTML = renderString;
+  formSubmitButton = seatsForm.querySelector('.seats_submit');
+  seatsForm.addEventListener('submit', submitSelectedTickets);
+  seatsForm.addEventListener('change', calculatePreOrder);
+}
+
+function prepareTicketsList() {
+  let renderString = '';
+  selectedSeats.forEach((item) => {
+    renderString += `<li class="pre_order__item">ряд ${item[0]}, место ${item[1]}</li>`;
+  });
+  ticketsList.innerHTML = renderString;
+}
+
+function calculatePreOrder() {
+  const checkedNodeList = seatsForm.querySelectorAll('input[type=checkbox]:checked:not(:disabled)');
+  if (checkedNodeList.length > 0) {
+    selectedSeats = [...checkedNodeList].map(({ id, dataset }) => [
+      parseInt(id.slice(4), 10),
+      parseInt(id.slice(id.indexOf('_') + 1), 10),
+      dataset.type,
+    ]);
+    prepareTicketsList();
+    auditorium.classList.add('auditorium-tight');
+    preOrder.classList.remove('pre_order-hidden');
+  } else if (checkedNodeList.length === 0) {
+    selectedSeats = [];
+    auditorium.classList.remove('auditorium-tight');
+    preOrder.classList.add('pre_order-hidden');
+  }
+}
+
+function confirmOrder() {
+  function payOrder() {
+    location.reload();
+  }
+
+  function cancelOrder() {
+    orderConfirmationWindow.classList.toggle('order_confirmation-hidden');
+    document.removeEventListener('keydown', modalKeyHandler);
+    confirmButton.removeEventListener('click', payOrder);
+    cancelButton.removeEventListener('click', cancelOrder);
+    formSubmitButton.focus();
+  }
+
+  function modalKeyHandler(e) {
+    if (e.key === 'Tab') {
+      if (e.shiftKey) {
+        if (document.activeElement === confirmButton) {
+          e.preventDefault();
+          cancelButton.focus();
+        }
+      } else {
+        if (document.activeElement === cancelButton) {
+          e.preventDefault();
+          confirmButton.focus();
+        }
+      }
+    }
+    if (e.key === 'Escape') {
+      cancelOrder();
+    }
```

Comment:
When modal is open, focus must go to the first focusable element in the modal then user iterate through the rest focusable elements.

Reply:
Well, when modal is open the first interactive element is focused by formSubmitButton.focus(); in line 110.
Since I only have two interactive elements on this modal I don't pick them in array or something, I simple chose this two buttons by querySelector's. When user press Tab on the second (and the last) button - focus goes to the first button, and when user press Shift+Tab on the first button - focus goes to the second (and the last) button. User can leave modal window by Escape or pressing (or clicking) some button, but not by pressing Tab/Shift+Tab.
And you can see this behavior of modal window in demo.
Or may be I just don't understand you.

Reply:
Well, when modal is open the first interactive element is focused by formSubmitButton.focus(); in line 110.
Since I only have two interactive elements on this modal I don't pick them in array or something, I simple chose this two buttons by querySelector's. When user press Tab on the second (and the last) button - focus goes to the first button, and when user press Shift+Tab on the first button - focus goes to the second (and the last) button. User can leave modal window by Escape or pressing (or clicking) some button, but not by pressing Tab/Shift+Tab.
And you can see this behavior of modal window in demo.
Or may be I just don't understand you.

Reply:
After last commit line 149 becomes line 172.
Prettier likes to wrap lines :)